### PR TITLE
Seller experience: Make options requirement aware of current site plan

### LINKED ISSUE #60710: Seller Experience: The simple path is not free.
We currently show the simple path (which uses Payments blocks) as being free, but to use a Payment block or any sort, you need to be on a paid plan. Let's duplicate the markup for the Woo path, but change it to "paid" (linking to the same place, the `/plans/` page.

Screen Shot 2022-02-01 at 7 49 24 AM

### PR DESCRIPTION
**Pull request by nelsonec87**
#### Changes proposed in this Pull Request

* Changed the simple path requirement (it's not free)
* Show a different message when the current site already fulfils the requirement.

#### Testing instructions

* Access `start/setup-site/intent?siteSlug=<SITE SLUG>&flags=seller-experience` (which enables the `seller-experience` feature flag)
* Click the "Start Selling" button
* Click "Continue"
* Check the new view
* With Free plan:
image

* Paid Plan:
image

* Business or eCommerce plans:
image

* The link should take you to the plans page.

Fixes #60710

### CODE REVIEW

**Comment by kwight** [comment]
Let's also handle the case when users have already paid for a plan – any paid plan for simple, the business plan for Woo: https://github.com/Automattic/wp-calypso/issues/60710#issuecomment-1027173235

**Comment by nelsonec87** [comment]
I've used a ternary operator inside the component markup because the logic is simple and it seems easy to read.
Does anyone feel differently about it?

```
{ isPaidPlan
	? translate( 'Included in your plan' )
	: translate( 'Requires a {{a}}paid plan{{/a}}', {
			components: {
				a: <a href={ `/plans/${ siteSlug }` } />,
			},
	} ) }
```

File: client/signup/steps/store-features/index.tsx [hunk 1]
```diff
@@ -28,6 +29,25 @@ export default function StoreFeaturesStep( props: Props ): React.ReactNode {
 	const subHeaderText = translate( 'Let’s create a website that suits your needs.' );
 	const siteSlug = props.signupDependencies.siteSlug;
 
+	const sitePlanSlug = useSelector( ( state ) => {
+		const site = getSite( state, siteSlug );
+		return site && site.plan && site.plan.product_slug;
```
Comment (kwight) [inline client/signup/steps/store-features/index.tsx]: I think you can use optional chaining here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

**Comment by kwight** [comment]
> I've used a ternary operator inside the component markup because the logic is simple and it seems easy to read.

Sure, looks good. 👍

File: client/signup/steps/store-features/index.tsx (hunk 1 above)
Comment (nelsonec87) [reply client/signup/steps/store-features/index.tsx]: Awesome! I'd forgotten about this one!

**Review by kwight — APPROVED** [review]
Tests Great! I tried three sites: free, Premium, and Business, and got the variations I expected. ✨ 

Just tidy up the chaining and sort the failed test, then 🚢

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7123738

Thank you @nelsonec87 for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7123738

Thank you @nelsonec87 for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7123738).

**Pull request merged**